source/slang/slang-mangle.cpp

```diff
@@ -547,6 +547,28 @@ namespace Slang
 
             for(auto paramDeclRef : parameters)
             {
+                // parameter modifier makes big difference in the spirv code generation, for example
+                // "out"/"inout" parameter will be passed by pointer. Therefore, we need to
+                // distinguish them in the mangled name to avoid name collision.
+                ParameterDirection paramDirection = getParameterDirection(paramDeclRef.getDecl());
+                switch (paramDirection)
+                {
+                case kParameterDirection_Ref:
+                    emitRaw(context, "ref");
+                    break;
+                case kParameterDirection_ConstRef:
+                    emitRaw(context, "constref");
+                    break;
+                case kParameterDirection_Out:
+                    emitRaw(context, "out");
+                    break;
+                case kParameterDirection_InOut:
+                    emitRaw(context, "inout");
+                    break;
+                default:
+                    emitRaw(context, "in");
```

Comment:
Should make a case for in, and default should be an abort error.

Reply:
It's no harm to do this, though I don't think it will go to default according to the implementation of `getParameterDirection`.